rgw/admin/user.go

```diff
@@ -0,0 +1,148 @@
+package admin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"errors"
+)
+
+// User is GO representation of the json output of a user creation
+type User struct {
+	UserID      string        `json:"user_id" url:"uid"`
+	DisplayName string        `json:"display_name" url:"display-name"`
+	Email       string        `json:"email" url:"email"`
+	Suspended   *int          `json:"suspended" url:"suspended"`
+	MaxBuckets  *int          `json:"max_buckets" url:"max-buckets"`
+	Subusers    []interface{} `json:"subusers"`
+	Keys        []struct {
+		User      string `json:"user"`
+		AccessKey string `json:"access_key" url:"access-key"`
+		SecretKey string `json:"secret_key" url:"secret-key"`
+	} `json:"keys"`
+	SwiftKeys []interface{} `json:"swift_keys"`
+	Caps      []struct {
+		Type string `json:"type"`
+		Perm string `json:"perm"`
+	} `json:"caps"`
+	OpMask              string        `json:"op_mask"`
+	DefaultPlacement    string        `json:"default_placement"`
+	DefaultStorageClass string        `json:"default_storage_class"`
+	PlacementTags       []interface{} `json:"placement_tags"`
+	BucketQuota         struct {
+		Enabled    *bool `json:"enabled"`
+		CheckOnRaw *bool `json:"check_on_raw"`
+		MaxSize    *int  `json:"max_size"`
+		MaxSizeKb  *int  `json:"max_size_kb"`
+		MaxObjects *int  `json:"max_objects"`
+	} `json:"bucket_quota"`
+	UserQuota struct {
+		Enabled    *bool `json:"enabled"`
+		CheckOnRaw *bool `json:"check_on_raw"`
+		MaxSize    *int  `json:"max_size"`
+		MaxSizeKb  *int  `json:"max_size_kb"`
+		MaxObjects *int  `json:"max_objects"`
+	} `json:"user_quota"`
+	TempURLKeys []interface{} `json:"temp_url_keys"`
+	Type        string        `json:"type"`
+	MfaIds      []interface{} `json:"mfa_ids"`
+	KeyType     string        `url:"key-type"`
+	Tenant      string        `url:"tenant"`
+	GenerateKey *bool         `url:"generate-key"`
+	PurgeData   *int          `url:"purge-data"`
```

Comment:
Why do these only have url annotations, while other fields have only json annotations?

Reply:
If api request parameters have a different structure than the response data, why do they share the same struct?

Reply:
For convenience and less confusion (despite the one writing the code maybe :p). Since the consumer needs to pass an exported type to build a request, it feels strange to return a different type, which ultimately shares many commonalities with the input type. Hope that makes sense.

Reply:
Yeah, let's say a bit. But my opinion is not strong enough that I would debate it more. 😊